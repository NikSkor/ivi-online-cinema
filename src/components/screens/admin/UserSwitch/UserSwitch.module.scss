@import '../../../../styles/vars.scss';

.title {
  text-transform: uppercase;
  font-weight: 600;
  font-size: 18px;
}

.toggleContainer {
  display: flex;
  gap: 20px;
  text-align: center;
  align-items: center;
  margin-bottom: 20px;
}

.toggle {
  display: inline-block;
  position: relative;
  width: 60px;
  height: 34px;
}

.toggleInput {
  height: 0;
  width: 0;
  opacity: 0;
}

.toggleSlider {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  border-radius: 34px;

  background-color: $red;
  transition: .5s;
}

.toggleSlider::before {
  content: '';
  position: absolute;
  cursor: pointer;
  left: 4px;
  bottom: 4px;
  height: 26px;
  width: 26px;
  border-radius: 50%;

  background-color: #fff;
  transition: .5s;
}

.toggleInput:focus+.toggleSlider {
  box-shadow: 0 0 1px $lightGray;
}

.toggleInput:checked+.toggleSlider {
  background-color: $secondary;
}

.toggleInput:checked+.toggleSlider::before {
  transform: translateX(26px);
}