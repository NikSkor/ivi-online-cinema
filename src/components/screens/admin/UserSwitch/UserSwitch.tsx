import { FC, useState } from "react";
import style from './UserSwitch.module.scss';
import { useAppDispatch } from "@/store/hooks";
import { newPage } from "@/store/slices/adminSlice";

interface ISwitch {
  isTrue: boolean,
  firstTitle: string,
  secondTitle: string,
  isGenres: Function
}

const UserSwitch: FC<ISwitch> = ({firstTitle, secondTitle, isTrue, isGenres}) => {

  const dispatch = useAppDispatch();

  const toggleHandler = () => {
    isTrue === true ? isTrue = false : isTrue = true;
    isGenres(isTrue);
    dispatch(newPage(1));
  }
  
  return (
    <div className={style.toggleContainer}>
          <h3 className={style.title}>{firstTitle}</h3>
          <label className={style.toggle} onChange={toggleHandler}>
            <input type="checkbox" className={style.toggleInput}/>
            <span className={style.toggleSlider}></span>
          </label>
          <h3 className={style.title}>{secondTitle}</h3>
        </div>
  )
}

export default UserSwitch;