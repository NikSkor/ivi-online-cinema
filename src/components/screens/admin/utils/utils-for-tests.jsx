import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import adminReducer from '../../../../store/slices/adminSlice';


export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { admin: adminReducer},
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}