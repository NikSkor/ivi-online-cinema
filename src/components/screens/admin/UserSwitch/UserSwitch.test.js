import { cleanup, screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import UserSwitch from '@/components/screens/admin/UserSwitch/UserSwitch';
import { renderWithProviders } from '../utils/utils-for-tests';

jest.mock('next/router', () => require('next-router-mock'));

describe('test UserSwitch component', () => {
  afterEach(cleanup);

  test('there is the checkbox', () => {
    renderWithProviders(
      <UserSwitch
        firstTitle={'Левый'}
        secondTitle={'Правый'}
        isTrue={true}
        isGenres={false}
      />
    );
    const chk = screen.getByRole('checkbox');
    expect(chk).toBeInTheDocument();
    expect(chk).toMatchSnapshot();
  });

  test('component renders props correctly, truthy values', () => {
    const tree = renderWithProviders(
      <UserSwitch
        firstTitle={'Левый'}
        secondTitle={'Правый'}
        isTrue={true}
        isGenres={false}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  test('component renders props correctly, falsy values', () => {
    const tree = renderWithProviders(
      <UserSwitch/>
    );
    expect(tree).toMatchSnapshot();
  });
});
