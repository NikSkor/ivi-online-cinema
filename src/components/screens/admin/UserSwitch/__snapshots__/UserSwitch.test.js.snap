// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test UserSwitch component component renders props correctly, falsy values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="toggleContainer"
      >
        <h3
          class="title"
        />
        <label
          class="toggle"
        >
          <input
            checked=""
            class="toggleInput"
            role="checkbox"
            type="checkbox"
          />
          <span
            class="toggleSlider"
          />
        </label>
        <h3
          class="title"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="toggleContainer"
    >
      <h3
        class="title"
      />
      <label
        class="toggle"
      >
        <input
          checked=""
          class="toggleInput"
          role="checkbox"
          type="checkbox"
        />
        <span
          class="toggleSlider"
        />
      </label>
      <h3
        class="title"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`test UserSwitch component component renders props correctly, truthy values 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="toggleContainer"
      >
        <h3
          class="title"
        >
          Левый
        </h3>
        <label
          class="toggle"
        >
          <input
            class="toggleInput"
            role="checkbox"
            type="checkbox"
          />
          <span
            class="toggleSlider"
          />
        </label>
        <h3
          class="title"
        >
          Правый
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="toggleContainer"
    >
      <h3
        class="title"
      >
        Левый
      </h3>
      <label
        class="toggle"
      >
        <input
          class="toggleInput"
          role="checkbox"
          type="checkbox"
        />
        <span
          class="toggleSlider"
        />
      </label>
      <h3
        class="title"
      >
        Правый
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`test UserSwitch component there is the checkbox 1`] = `
<input
  class="toggleInput"
  role="checkbox"
  type="checkbox"
/>
`;
